HeterogeneousCore/CUDAUtilities/interface/HistoContainer.h

```diff
@@ -86,15 +82,20 @@ namespace cms {
       uint32_t *off = (uint32_t *)((char *)(h) + offsetof(Histo, off));
       size_t wss = Histo::wsSize();
       assert(wss > 0);
-      CubDebugExit(cub::DeviceScan::InclusiveSum(ws, wss, off, off, Histo::totbins(), stream));
+      auto nthreads = 512+256;
+      auto nblocks = (Histo::totbins() + nthreads - 1) / nthreads;
+      assert(nblocks<=1024);
```

Comment:
where does this limitation come from ?

Reply:
good point.
It was due to the size of the shared array that now is not fixed anymore

Reply:
ok
do we still have a limit on the number of blocks, either hard-coded or dynamic ?

Reply:
the limit still exists due to the limit in blockPrefixScan
it is asserted in there though